examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a1b589c0",
+   "metadata": {},
+   "source": [
+    "Another factor we have not discussed is the step-wise execution time for the different optimizers. I now pull out this information saved during experiment:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "508c0d73",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "QNSPSA takes 0.43 s per step\n",
+      "SPSA takes 0.21 s per step\n",
+      "QNG takes 4.74 s per step\n",
+      "GD takes 4.16 s per step\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f'QNSPSA takes %.2f s per step' % (results['qnspsa_duration'] / spsa_repeats / max_iter))\n",
+    "print(f'SPSA takes %.2f s per step' % (results['spsa_duration'] / spsa_repeats / max_iter))\n",
+    "print(f'QNG takes %.2f s per step' % (results['qng_duration'] / max_iter))\n",
+    "print(f'GD takes %.2f s per step' % (results['gd_duration'] / max_iter))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7199f820",
+   "metadata": {},
+   "source": [
+    "We can see the obvious advantage of the stochasitc optimizers (SPSA and QN-SPSA) in their execution time. Note that there are only 4 parameters in a depth-2 QAOA ansatz, yet it takes 10X more time for the gradient-based optimizers to finish a single optimization step, comparing to QN-SPSA (20X comparing to SPSA). The reason is that estimating the gradient with parameter shift rule requires to run 2$n_G$ circuits, where $n_G$ is the number of gates. The QAOA ansatz has limited parameters, but a rather large amount of gates applied. On the other hand, stochastic methods always execute a fixed number of circuits for each step. Combining the execution time observation with the optmization result, we can then clearly see the advantage of QN-SPSA in this use case."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ccdb1c5a",
+   "metadata": {},
+   "source": [
+    "Just as an additional sanity check, I also reproduce the experiment result in Fig 1(b) from Ref. [1] (Refer to Ref. [1] for the definition of the problem Hamiltonian and the ansatz). The  script `benchmark_ref_paper_converge_speed.py` for training, and the following two paragraphs take 70 min to finish."
```

Comment:
It would be good to add a sentence about what the problem described in `benchmark_ref_paper_converge_speed.py` is.